godtools/App/Features/Menu/Domain/UseCases/DeleteAccountUseCase/DeleteAccountUseCase.swift

```diff
@@ -0,0 +1,50 @@
+//
+//  DeleteAccountUseCase.swift
+//  godtools
+//
+//  Created by Levi Eggert on 5/23/23.
+//  Copyright © 2023 Cru. All rights reserved.
+//
+
+import Foundation
+import Combine
+
+class DeleteAccountUseCase {
+    
+    private let userAuthentication: UserAuthentication
+    private let userDetailsRepository: UserDetailsRepository
+    private let mobileContentAuthTokenRepository: MobileContentAuthTokenRepository
+    
+    init(userAuthentication: UserAuthentication, userDetailsRepository: UserDetailsRepository, mobileContentAuthTokenRepository: MobileContentAuthTokenRepository) {
+        
+        self.userAuthentication = userAuthentication
+        self.userDetailsRepository = userDetailsRepository
+        self.mobileContentAuthTokenRepository = mobileContentAuthTokenRepository
+    }
+    
+    func deleteAccountPublisher() -> AnyPublisher<Void, URLResponseError> {
+                
+        return userDetailsRepository.deleteAuthorizedUserDetails()
+            .flatMap({ (void: Void) -> AnyPublisher<Void, URLResponseError> in
+                                
+                return self.userAuthentication.signOutPublisher()
+                    .mapError { (error: Error) in
+                        return .otherError(error: error)
+                    }
+                    .eraseToAnyPublisher()
+            })
+            .flatMap({ (void: Void) -> AnyPublisher<Void, Never> in
+                        
+                self.mobileContentAuthTokenRepository.deleteCachedAuthToken()
+                
+                return SetTimeout.setTimeoutPublisher(seconds: 1)
```

Comment:
What does this timeout do?  Make sure the auth token gets removed from the cache before returning?

Reply:
Ah this was because I would like the UI to display a little longer here in the event the api request is really fast.  Not sure about having this in the UseCase though.  I'm going to look to moving this delay to the Flow with a combination of elapsed time.

Reply:
Hmm, I might be able to use debounce or delay.

https://developer.apple.com/documentation/combine/publisher/debounce(for:scheduler:options:)

https://developer.apple.com/documentation/combine/publisher/delay(for:tolerance:scheduler:options:)